tensorrt_llm/commands/serve.py

```diff
@@ -332,6 +346,80 @@ def serve(
     launch_server(host, port, llm_args, metadata_server_cfg, server_role)
 
 
+@click.command("mmemb_serve")
+@click.argument("model", type=str)
+@click.option("--host",
+              type=str,
+              default="localhost",
+              help="Hostname of the server.")
+@click.option("--port", type=int, default=8000, help="Port of the server.")
+@click.option('--log_level',
+              type=click.Choice(severity_map.keys()),
+              default='info',
+              help="The logging level.")
+@click.option("--max_batch_size",
+              type=int,
+              default=BuildConfig.max_batch_size,
+              help="Maximum number of requests that the engine can schedule.")
+@click.option("--gpus_per_node",
+              type=int,
+              default=None,
+              help="Number of GPUs per node. Default to None, and it will be "
+              "detected automatically.")
+@click.option("--trust_remote_code",
+              is_flag=True,
+              default=False,
+              help="Flag for HF transformers.")
+@click.option(
+    "--extra_encoder_options",
+    type=str,
+    default=None,
+    help=
+    "Path to a YAML file that overwrites the parameters specified by trtllm-serve."
+)
+@click.option("--metadata_server_config_file",
+              type=str,
+              default=None,
+              help="Path to metadata server config file")
+def serve_encoder(model: str, host: str, port: int,
```

Comment:
We should have integration tests that check this `mm_embeddding_serve` is working as expected.

Reply:
Added an integration test `test_openai_mmencoder_example` (_test_openai_mmencoder.py)